src/main/java/teammates/sqllogic/core/FeedbackResponsesLogic.java

```diff
@@ -184,4 +197,51 @@ public boolean areThereResponsesForQuestion(UUID questionId) {
     public boolean hasResponsesForCourse(String courseId) {
         return frDb.hasResponsesForCourse(courseId);
     }
+
+    /**
+     * Updates a feedback response.
+     *
+     * <p>Cascade updates its associated feedback response comment
+     * (e.g. associated response ID, giverSection and recipientSection).
+     *
+     * <p>If the giver/recipient field is changed, the response is updated by recreating the response
+     * as question-giver-recipient is the primary key.
+     */
+    public FeedbackResponse updateFeedbackResponseCascade(FeedbackResponse feedbackResponse)
+            throws InvalidParametersException, EntityDoesNotExistException, EntityAlreadyExistsException {
+        FeedbackResponse oldResponse = frDb.getFeedbackResponse(feedbackResponse.getId());
+        FeedbackResponse newResponse = frDb.updateFeedbackResponse(feedbackResponse);
+
+        boolean isResponseIdChanged = !oldResponse.getId().equals(newResponse.getId());
```

Comment:
Always false if new response is created. This is a randomly generated uuid, not the same as the old id.